### DISCUSSION TITLE
Use .mkv attached covers?

### DISCUSSION THREAD

**Discussion by StormBytePP** [gerbera/gerbera#3011] [Q&A] [answered]
Can Gerbera use cover thumbnails attached to mkv files instead of using ffmpegthumbnail to generate them? Or perhaps any option to specify this?

Currently I am using miniDLNA and want to replace it by Gerbera yet this issue is blocking me from doing so as I already maintain my collection with everything having cover thumb in the video files.

Thanks.

**Comment by KarlStraussberger** [comment, marked as answer]
Yes, we implemented artwork support. As matroska is a very versatile format, it depends on the encoder. Anyway, you can set the order of the thumbnail source.

**Reply by StormBytePP** [reply]
Thank you, any hint about that thumbnail source order?

**Reply by KarlStraussberger** [reply]
See here https://docs.gerbera.io/en/latest/config-import.html#order

HINT: You should browse through configuration docs, before starting your first import.

**Reply by StormBytePP** [reply]
Yes, thanks for the link but the reason I asked for a hint is because I feel confused towards it, for example, in the upper document it says as example:
1: `
<resources>
     <order/>
     <fanart/>
     <subtitle/>
     <metafile/>
     <resource/>
     <container/>
   </resources>
`
There, order is an empty tag but following order definition states it is a content-tag like 2: `<order>...</order>`

So I feel confused, is it an empty tag indicating that the following items are ordered like 1 or is it a content tag having the rest of tags inside like 2?

**Reply by KarlStraussberger** [reply]
The upper one just shows the structure of the section. You should start with `gerbera --create-config`.

My order section looks like
```
<order>
  <handler name="Default"/>
  <handler name="TagLib"/>
  <handler name="FFmpegThumbnailer"/>
  <handler name="Fanart"/>
</order>
```
to get thumbnailer values before fanart.
